solana/programs/ntt-quoter/src/processor/request_relay.rs

```diff
@@ -29,10 +28,18 @@ pub struct RequestRelay<'info> {
     )]
     pub registered_chain: Account<'info, RegisteredChain>,
 
+    #[account(
+        seeds = [
+            RegisteredNtt::SEED_PREFIX,
+            (&outbox_item as &dyn AsRef<AccountInfo<'info>>).as_ref().owner.to_bytes().as_ref()
+        ],
+        bump = registered_ntt.bump,
+    )]
+    pub registered_ntt: Account<'info, RegisteredNtt>,
+
     //TODO eventually drop the released constraint and instead implement release by relayer
     #[account(
-        owner = example_native_token_transfers::ID,
```

Comment:
Is there any other ownership check or permission that we should be checking here?
My initial thinking is no because this is a relayer so it does not need any sort of authorization. But I still want to raise it in case we are missing something

Reply:
My main concern when writing this was actually that Anchor might check the owner, even though I don't intend for it to do so:
https://docs.rs/anchor-lang/latest/anchor_lang/accounts/account/struct.Account.html

In fact, when I implemented it the way it is right now, I missed that `OutboxItem` actually has the Anchor `#[account]` attribute and thought it was just a normal struct with some derives... (without the account attribute, it wouldn't implement the owner trait and so we'd be in the clear)

So this is actually buggy because it will introduce this unwanted owner check, which is what might have made you subconsciously feel uneasy when looking at this!

Reply:
The original implementation was already bullshit / a red herring because `Account` checks that the owner is `T::owner()`. For some reason, I thought it would check that the owner is my current program without the `owner = ...` but that's false.

Reply:
Ah OK, I assumed that this was a temporary hard-coding similar to the WORMHOLE_TRANSCEIVER_INDEX.

So basically this should be removed (like you have it in this PR) and Anchor should take care of it like normal. Is that your understanding too?

Reply:
What I need is something that deserializes the account (and checks the discriminator) but without checking the account owner (because I don't know or care which ntt program it is). I don't think Anchor provides this functionality (because it is a bit of a weird Frankenstein) so I'm currently looking into what the best way is to accomplish this.